### PR TITLE
Don't display invalid shaders

### LINKED ISSUE #1468: The "Shader Packs" menu of an instance shows the .txt file for a shader.
### Operating System

Windows 11

### Version of Prism Launcher

Prism Launcher 7.2

### Version of Qt

Qt: 6.5.1

### Description of bug

The "shader packs" menu of an instance displays the .txt configuration file for a shader and treats it as a separate shader. Instead, the "shader packs" menu of an instance should display only shaders. One way to fix this is to check the directory of the shaders for the instance, and if there is a .zip file or a folder, make the launcher check for a folder called "shaders" inside the .zip file or the folder, and only show the .zip file or the folder if it has that folder inside it (the "shaders" folder should be the only folder inside the .zip file or the folder).
Also, if what I am talking about will be fixed, then also make it so that if a shader has a .txt configuration file, and the user uses the shader packs menu to disable/remove a shader, then the launcher should also disable/remove the .txt file (if it exists) for that shader and bring it back if the user re-enables the shader.

See the ".zip.txt" file for the shader:
Screenshot 2023-07-31 195612

### Steps to reproduce

1. Have a shader in the "shaderpacks" folder of an instance, and also a .txt configuration file for the shader.
2. Open prism launcher.
3. Press "Edit" on the instance.
4. Go to "Shader packs." 
5. You will see both the shader and its .txt configuration file.

### Suspected cause

The launcher displays everything in the shaderpacks directory and doesn't check for shaders specifically.

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

### PR DESCRIPTION
**Pull request by Trial97**
fixes #1468

### CODE REVIEW

**Comment by TheKodeToad** [comment]
This is inconsistent with other pages where invalid files will display. This behaviour might be logical because of the way configuration works though.

**Comment by Trial97** [comment]
I know that the behavior is inconsistent but shaders have the configuration in the same folder as the shader resulting in adding the file to the list as a shader.
There is no way around implementing this other than filtering out invalid files.
On other pages we expect to have the same resource type: so we display all valid and invalid ones.

**Review by TheKodeToad — APPROVED** [review]

File: launcher/minecraft/mod/ShaderPackFolderModel.h [hunk 1]
```diff
@@ -9,4 +12,14 @@ class ShaderPackFolderModel : public ResourceFolderModel {
...
+
+    [[nodiscard]] Task* createUpdateTask() override
+    {
+        return new BasicFolderLoadTask(m_dir, [](QFileInfo const& entry) { return makeShared<ShaderPack>(entry); });
+    }
+
+    [[nodiscard]] Task* createParseTask(Resource& resource) override
+    {
+        return new LocalShaderPackParseTask(m_next_resolution_ticket, static_cast<ShaderPack&>(resource));
+    }
```
Comment (Scrumplex) [inline launcher/minecraft/mod/ShaderPackFolderModel.h:24]: How are these involved here? Why wasn't this needed before? Just wondering, this shouldn't block merging

**Review by Scrumplex — APPROVED** [review]

File: launcher/minecraft/mod/ShaderPackFolderModel.h (hunk 1 above)
Comment (TheKodeToad) [reply launcher/minecraft/mod/ShaderPackFolderModel.h:24]: it allows using a subclass of Resource and therefore overriding `applyFilter`

Comment (Trial97) [reply launcher/minecraft/mod/ShaderPackFolderModel.h:24]: TheKobeToad explanation is accurate. 
As for why was not used before, I'm not sure as it only looks like the parsing code was used for mods.
There are similar unused tasks for other resources.
The only place it is used is when calling a function to determine the type of a resource( used in curseforge modpack import)

**Pull request merged**